src/components/PopupWithForm.js

```diff
@@ -5,20 +5,32 @@ class PopupWithForm extends Popup {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
   }
 
   _getInputValues() {
     const inputList = this._popupForm.querySelectorAll(".modal__input");
     const formValues = {};
 
     inputList.forEach((input) => (formValues[input.name] = input.value));
+    console.log(formValues);
 
     return formValues;
   }
 
+  setButtonText(submit, buttonText = "Saving...") {
+    if (submit) {
+      this._submitButton.textContent = buttonText;
+    } else {
+      this._submitButton.text;
```

Comment:
* When I submit any form the button text gets `Saving...` for ever https://disk.yandex.ru/i/KUFAqtMoys3T3Q 


`this._submitButtonText;` should be used here

```js

    } else {
      this._submitButton.textContent = this._submitButtonText;
    }
```

Reply:
completed.